eng/versioning/external_dependencies.txt

```diff
@@ -181,7 +181,7 @@ cosmos_io.netty:netty-handler;4.1.45.Final
 cosmos_io.netty:netty-handler-proxy;4.1.45.Final
 cosmos_io.netty:netty-tcnative;2.0.27.Final
 cosmos_io.netty:netty-transport-native-epoll;4.1.45.Final
-cosmos_io.projectreactor:reactor-core;3.3.3.RELEASE
+cosmos_log4j:log4j;1.2.17
```

Comment:
@kushagraThapar We shouldn't be using this version of log4j. This particular version went out of support in 2015 and @conniey had updated these dependencies in early Feb as part of this [PR](https://github.com/Azure/azure-sdk-for-java/pull/8031). I would like to see this cosmos specific version updated to use the 2.x common versions above.

Reply:
@JimSuplizio  - the only reason we had to add log4j is because of this dependency. 
```
<dependency>
      <groupId>com.fasterxml.uuid</groupId>
      <artifactId>java-uuid-generator</artifactId>
      <version>3.2.0</version> <!-- {x-version-update;com.fasterxml.uuid:java-uuid-generator;external_dependency} -->
    </dependency>
```
Cosmos doesn't use this dependency of log4j, but uuid does, and it was complaining for `log4j module not found, required by com.fasterxml.uuid` 
So that is the reason we had to add this.

Reply:
Is it possible to move to version 4? It uses slf4j, which would align with our underlying logging framework.

https://github.com/cowtowncoder/java-uuid-generator/blob/master/README.md#dependencies

Reply:
@conniey yes that should be possible, I will give it a shot.

Reply:
Updated the version here: https://github.com/Azure/azure-sdk-for-java/pull/9757